pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2973,10 +2975,12 @@ type CephBlockPoolRadosNamespaceList struct {
 // CephBlockPoolRadosNamespaceSpec represents the specification of a CephBlockPool Rados Namespace
 type CephBlockPoolRadosNamespaceSpec struct {
 	// The name of the CephBlockPoolRadosNamespaceSpec namespace. If not set, the default is the name of the CR.
+	// +kubebuilder:validation:XValidation:message="CephBlockPoolRadosNamespaceSpec name is immutable",rule="self == oldSelf"
 	// +optional
 	Name string `json:"name,omitempty"`
 	// BlockPoolName is the name of Ceph BlockPool. Typically it's the name of
 	// the CephBlockPool CR.
+	// +kubebuilder:validation:XValidation:message="CephBlockPoolRadosNamespaceSpec BlockPoolName is immutable",rule="self == oldSelf"
```

Comment:
I'd suggest no need to include the type name in the message, and let's use the lower case yaml name
```suggestion
	// +kubebuilder:validation:XValidation:message="blockPoolName is immutable",rule="self == oldSelf"
```

Reply:
updated